test/integration/product-import.spec.coffee

```diff
@@ -189,6 +189,36 @@ describe 'Product Importer integration tests', ->
     .then (result) ->
       service.byId(id).delete(result.body.version)
 
+  it 'should handle an empty import', (done) ->
+    @import.performStream([], -> {})
+    .then =>
+      @client.productProjections.staged(true)
+      .all()
+      .where("productType(id=\"#{@productType.id}\")")
+      .fetch()
+    .then (result) ->
+      expect(result.body.results.length).toBe(0)
+      done()
+
+  it 'should import products even when they do not have SKUs', (done) ->
```

Comment:
@hisabimbola - @butenkor is correct - Current implementation takes all SKUs from all variants, fetches existing products and calculates diffs.

That means that updating products which do not have SKUs is not possible.